packages/core-components/src/components/Link/Link.tsx

```diff
@@ -46,12 +47,44 @@ const useStyles = makeStyles(
 
 export const isExternalUri = (uri: string) => /^([a-z+.-]+):/.test(uri);
 
+/**
+ * Get the app base path from the configured app baseUrl.
+ *
+ * The returned path does not have a trailing slash.
+ */
+const getBasePath = (configApi: Config) => {
+  let { pathname } = new URL(
+    configApi.getOptionalString('app.baseUrl') ?? '/',
+    'http://dummy.dev', // baseUrl can be specified as just a path
+  );
+  pathname = pathname.replace(/\/*$/, '');
```

Comment:
Perhaps we can use `trimEnd(pathname, '/') ` from lodash instead as codeql sometimes complain about regexes.